proposals/native-lock.md

```diff
@@ -0,0 +1,83 @@
+# Native lock
+
+## Summary
+[summary]: #summary
```

Comment:
It's in many proposals I believe; it's part of [the template](https://github.com/dotnet/csharplang/blob/main/proposals/proposal-template.md?plain=1).

It defines a link `[summary]` to point to `#summary`, so it can be then used like `see [the summary section][summary]` where in this case it's not really better than using just `see [the summary section](#summary)` but I guess if the `## Summary` heading changes, we don't have to update all links, just this one. [Some docs](https://www.markdownguide.org/basic-syntax/#reference-style-links).